espnet2/asr/frontend/adapters_utils/adapters/adapter_transformer.py

```diff
@@ -0,0 +1,108 @@
+import torch
+import torch.nn as nn
+from fairseq.models.wav2vec.wav2vec2 import TransformerSentenceEncoderLayer
+
+from espnet2.asr.frontend.adapters_utils.adapters.adapter import Adapter
+
+
+class AdapterTransformerSentenceEncoderLayer(TransformerSentenceEncoderLayer):
+    """
+    Modified fairseq's TransformerSentenceEncoderLayer for wav2vec2 with adapters.
+    Link:
+    https://github.com/
+    facebookresearch/fairseq/blob/
+    06c65c8297396959c025bf61aaeaa6f4ace9940e/
+    fairseq/models/wav2vec/wav2vec2.py#L1175
+    """
+
+    def __init__(
+        self,
+        embedding_dim: float = 768,
+        ffn_embedding_dim: float = 3072,
+        num_attention_heads: int = 8,
+        dropout: float = 0.1,
+        attention_dropout: float = 0.1,
+        activation_dropout: float = 0.1,
+        activation_fn: str = "relu",
+        layer_norm_first: bool = False,
+        adapter_down_dim: int = 192,
+    ) -> None:
+        super().__init__(
+            embedding_dim,
+            ffn_embedding_dim,
+            num_attention_heads,
+            dropout,
+            attention_dropout,
+            activation_dropout,
+            activation_fn,
+            layer_norm_first,
+        )
+        self.adapter1 = Adapter(embedding_dim, adapter_down_dim)
+        self.adapter2 = Adapter(embedding_dim, adapter_down_dim)
+
+    def forward(
+        self,
+        x: torch.Tensor,
+        self_attn_mask: torch.Tensor = None,
+        self_attn_padding_mask: torch.Tensor = None,
+        need_weights: bool = False,
+        att_args=None,
+    ) -> torch.Tensor:
```

Comment:
Ditto. It's better to add some the shape information of each input tensor. It may help users to debug.